lib/elixir/lib/kernel/typespec.ex

```diff
@@ -595,6 +595,10 @@ defmodule Kernel.Typespec do
     Enum.reduce Enum.reverse(args), fn(arg, expr) -> {:|, [line: line], [arg, expr]} end
   end
 
+  defp typespec_to_ast({:type, line, :function, args}) do
+    typespec_to_ast({:type, line, :fun, args})
```

Comment:
`function()` is equivalent to `fun()` but not an `atom()`, I don't think we should support it as there is no benefit for having `function()` then, as `fun()` is already used and also shorter.